buildSrc/src/main/java/org/elasticsearch/gradle/Jdk.java

```diff
@@ -111,6 +111,15 @@ public TaskDependency getBuildDependencies() {
         return configuration.getBuildDependencies();
     }
 
+    public Object getLazyPath() {
```

Comment:
I don't think this is necessary. The `toString()` method of `Jdk` delegates to `getPath()` already so you could just pass the `Jdk` instance itself anywhere you'd need this lazy evaluation.

Reply:
This actually didn't work as it was written before, I'm not sure how I didn't catch it before, it's probably a combination of really early in the morning and being in an airport ... 

The problem here is that GStrings passed to `test.executable` bellow do get evaluated ( even if they have a closure ) before they are supposed to, resolving the configuration too early. Passing in the JDK is evaluated as it should, but doesn't work as it points to java home missing the  `bin/java` part.